### ISSUE TITLE
"IDE0054: Use '++' operator" appears in inappropriate contexts

### ISSUE THREAD

**Issue by nivekh** [dotnet/roslyn#72422]
**Brief description:**

When writing a line of code of the pattern

`intVar = intVar + 1 % x`

where `%` is another operator and `x` is another operand, the suggestion "IDE0054: Use '++' operator" appears.
![image](https://github.com/dotnet/roslyn/assets/115717794/eb2eda10-c6cc-4821-9181-8b875ea0c07b)

This doesn't really make sense, as this would require breaking the operation up into two separate writes; `intVar = intVar++ % x` will write to `intVar` to increment it, and then write again to apply the modulo operation.

**Languages applicable:**

C#

**Code example that the analyzer should report:**
```
int colorIndex = 0; 
colorIndex = colorIndex + 1 % 2;
```
**Additional information:**

Please let me know if this isn't the correct route for this issue report.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by sharwell** [comment]
I believe this is not a bug. The analyzer takes this code:

```cs
int colorIndex = 0; 
colorIndex = colorIndex + 1 % 2;
```

And suggests converting it to this code:

```cs
int colorIndex = 0;
colorIndex++;
```

The deduction process for this is as follows:

1. The `%` operator has precedence over the `+` operator. `1 % 2` folds to the constant `1`.
2. After constant folding, the line is equivalent to `colorIndex = colorIndex + 1`.
3. The previous line is further simplified to `colorIndex++`, which is what the code fix applies.

**Comment by CyrusNajmabadi** [comment]
Put another way, your original code is parsed as if it was `colorIndex = colorIndex + (1 % 2);` not `colorIndex = (colorIndex + 1) % 2;`.  The analyzer's suggestion and change is correct given that actual interpretation.  I'm going to close this out.

**Comment by nivekh** [comment]
I have updated my example to be a bit less minimal and demonstrate that this analyser still triggers in a non-trivial context (i.e. within a loop where the modulo will actually roll over multiple times)

**Comment by CyrusNajmabadi** [comment]
`1%3` is still `1` in your updated example.

**Comment by nivekh** [comment]
Sorry, missed the operator precedence issue and then realized my mistake right after leaving that comment, you're right.

**Comment by CyrusNajmabadi** [comment]
No worries :)
